adlfs/spec.py

```diff
@@ -2153,9 +2154,22 @@ def _get_chunks(self, data):
         length = len(data)
         while start < length:
             end = min(start + self.blocksize, length)
-            yield data[start:end]
+            yield start, end
             start = end
 
+    async def _stage_block(self, data, start, end, block_id, semaphore):
+        if self._sdk_supports_memoryview_for_writes():
+            # Use memoryview to avoid making copies of the bytes when we splice for partitioned uploads
+            data = memoryview(data)
```

Comment:
Let's wrap the data in a memoryview after we've entered the semaphore context so we upload the property that no logic is run until there is availability in the semaphore as we are effectively treating this method as a worker now.